pkg/px-resource-gateway/priority_queue.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	pb "github.com/libopenstorage/operator/proto"
+)
+
+var (
+	priorityDescendingOrder = []pb.SemaphoreAccessPriority_Type{
+		pb.SemaphoreAccessPriority_HIGH,
+		pb.SemaphoreAccessPriority_MEDIUM,
+		pb.SemaphoreAccessPriority_LOW,
+	}
+)
+
+type PriorityQueue interface {
+	Enqueue(clientId string, priority pb.SemaphoreAccessPriority_Type)
+	Dequeue(priority pb.SemaphoreAccessPriority_Type)
+	Front() (string, pb.SemaphoreAccessPriority_Type)
+	Remove(clientId string)
+}
+
+type priorityQueue struct {
+	sync.Mutex
+	// using list.List for effecicient implementation of queues
+	queues map[pb.SemaphoreAccessPriority_Type]*list.List
+}
+
+func NewPriorityQueue() PriorityQueue {
+	queues := map[pb.SemaphoreAccessPriority_Type]*list.List{}
```

Comment:
style: normally `queues := make(map[pb.SemaphoreAccessPriority_Type]*list.List)`, but what you have works.